src/qt/test/wallettests.cpp

```diff
@@ -313,6 +391,76 @@ void TestGUI(interfaces::Node& node)
     QCOMPARE(walletModel.wallet().getAddressReceiveRequests().size(), size_t{0});
 }
 
+void TestGUIWatchOnly(interfaces::Node& node, TestChain100Setup& test)
+{
+    const std::shared_ptr<CWallet>& wallet = SetupLegacyWatchOnlyWallet(node, test);
+
+    // Create widgets and init models
+    std::unique_ptr<const PlatformStyle> platformStyle(PlatformStyle::instantiate("other"));
+    MiniGUI mini_gui(node, platformStyle.get());
+    mini_gui.initModelForWallet(node, wallet, platformStyle.get());
+    WalletModel& walletModel = *mini_gui.walletModel;
+    SendCoinsDialog& sendCoinsDialog = mini_gui.sendCoinsDialog;
+
+    // Update walletModel cached balance which will trigger an update for the 'labelBalance' QLabel.
+    walletModel.pollBalanceChanged();
+    // Check balance in send dialog
+    CompareBalance(walletModel, walletModel.wallet().getBalances().watch_only_balance,
+                   sendCoinsDialog.findChild<QLabel*>("labelBalance"));
+
+    // Set change address
+    sendCoinsDialog.getCoinControl()->destChange = GetDestinationForKey(test.coinbaseKey.GetPubKey(), OutputType::LEGACY);
+
+    // Time to reject "save" PSBT dialog ('SendCoins' locks the main thread until the dialog receives the event).
+    QTimer timer;
+    timer.setInterval(500);
+    QObject::connect(&timer, &QTimer::timeout, [&](){
+        for (QWidget* widget : QApplication::topLevelWidgets()) {
+            if (widget->inherits("QMessageBox")) {
+                QMessageBox* dialog = qobject_cast<QMessageBox*>(widget);
+                QAbstractButton* button = dialog->button(QMessageBox::Discard);
+                button->setEnabled(true);
+                button->click();
+                timer.stop();
+                break;
+            }
+        }
+    });
+    timer.start(500);
```

Comment:
It seems all these lines of code might be replaced with `ConfirmMessage(nullptr, 500ms);`.

Reply:
> It seems all these lines of code might be replaced with `ConfirmMessage(nullptr, 500ms);`.

That should trigger the filesystem window to dump the psbt info to a file, which would incur in another step to cancel the dialog. The idea here is to obtain the data through the clipboard only (see few lines below), not through a file.

That being said, agree that we could move this into the utility file in the future and cleanup some code.

Reply:
> That will trigger the filesystem window to dump the psbt info to a file...

Why? The `ConfirmMessage` triggers the default button, which is `QMessageBox::Discard`, no?

Reply:
> > That will trigger the filesystem window to dump the psbt info to a file...
> 
> Why? The `ConfirmMessage` triggers the default button, which is `QMessageBox::Discard`, no?

Yeah ok, I did not see the manual `setDefaultButton` call. That will work fine while the dialog takes less than 500ms in popping up. If it takes longer, the test will stall because `ConfirmMessage` executes runs the time only once (`QTimer::singleShot`).